### ISSUE TITLE
Configuration file should be stored in the platform-specific config location

### ISSUE THREAD

**Issue by brollb** [enigmasys/enigma#164]
Currently, the CLI is configured via an `application.yml` file stored in the same directory as the jar. This isn't great since applications are usually installed in some shared system location. This results in two issues:
- the logged in user likely does not have write permissions and, as a result, cannot configure the application
- multiple users cannot have their own configurations

Alternatively, the configuration should be stored in the platform-specific configuration directory. For example, on linux, I would expect it to be saved in `~/.config/leap_cli/config.yml`. I know there are libraries to make tasks like these easier (like [confy](https://github.com/rust-cli/confy)). It would be worth checking into similar libraries for Kotlin.

**Comment by yogeshVU** [comment]
Agreed. Having a platform-specific configuration directory will be nice. 
Currently, the `application.yml` can be located in the same directory as the jar. One can pass a custom `application.yml` as an argument to the jar with the location of the directory path to the invocation command something like:
`java -jar app.jar --spring.config.location=file:///Users/home/config/application.yml`
